Moveset.hpp

```diff
@@ -34,10 +34,49 @@ class Moveset {
 
     // If right side has an offet ret doesn't push it back to ret's offsets
     for (auto a : ret.offsets) {
+      //ugly bool because goto's are frowned upon
+      bool push = true;
       for ( auto b : m.offsets ) {
-	if ( a != b )
-	  ret.offsets.push_back(b);
+	if ( a == b ){
+          push=false;
+          break;
+        }
       }
+      if(push)
+        ret.offsets.push_back(a);
+    }
+    return ret;
+  }
+
+  Moveset operator^(const Moveset& m ) {
+    Moveset ret;
+
+    // Just XOR the flags
+    ret.horizontal = (this->horizontal ^ m.horizontal);
+    ret.vertical = (this->vertical ^ m.vertical);
+    ret.diagonal = (this->diagonal ^ m.diagonal);
+    ret.circular = (this->circular ^ m.circular);
+
+    
+    // If right side has an offet ret doesn't push it back to ret's offsets
+    for (auto a : offsets) {
+      for ( auto b : m.offsets ) {
+      	if ( a == b )
+          //Will they notice?
+          goto continueOuterLoop;
+      }
+      ret.offsets.push_back(a);
+      continueOuterLoop:
+      continue;
+    }
+    for (auto a : m.offsets) {
+      for ( auto b : offsets ) {
+      	if ( a == b )
+          goto continueSecondOuterLoop;
```

Comment:
🙄

Reply:
What would you suggest?

Reply:
Ignoring the stigma of the GOTO, it is performing exactly the same as a CONTINUE here.
Its just a named continue. :eyes: